src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +33,14 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                return this.ExpectedValue.GetType().Name;

```

Comment:
![image](https://cloud.githubusercontent.com/assets/677704/23332323/2819f990-fb78-11e6-8afd-d53e2aa0cf23.png)


But I do know that we had no test enforcing it! I'll add one, and alter the fix.